projects/layer-control/src/lib/layerentry/layerentry.component.ts

```diff
@@ -110,18 +122,32 @@ export class LayerentryComponent implements OnInit {
         this.layersSvc.updateLayer(selectedLayer, selectedLayer.filtertype || 'Layers'); // TODO check for baselayers!!!!!!
       }
     } else {
-      /** "radio" for Baselayers */
-      if (group.filtertype === 'Baselayers') {
-        for (const layer of group.layers) {
-          layer.visible = layer === selectedLayer;
+      if (group.layers.length > 0) {
+        /** "radio" for Baselayers */
+        if (group.filtertype === 'Baselayers') {
+          for (const layer of group.layers) {
+            layer.visible = layer === selectedLayer;
+          }
+          this.update.emit({
+            layer: this.layer
+          });
+          /** "checkbox" for all other layers */
+        } else {
+          const tempGroupVisible = group.visible;
+          /** change visibility of the selected layer */
+          selectedLayer.visible = !selectedLayer.visible;
+
+          /** check if group visibility has changed */
+          if (tempGroupVisible !== group.visible) {
```

Comment:
Can the group visibility change here at all? Setting the `selectedLayer`s visibility does not trigger any side effects, does it?

Reply:
The selected layer is in the group array, therefore if i change the visibility of it also the visibility of the group can change.